lib/modules/datasource/metadata.ts

```diff
@@ -131,8 +137,11 @@ export function addMetaData(
     if (parsedHomePage?.hostname.includes('github')) {
       if (!dep.sourceUrl) {
         dep.sourceUrl = dep.homepage;
+        if (urlPathDepth(dep.homepage) < 2) {
+          // remove homepage if its not a link to a path in a github repo.
+          delete dep.homepage;
+        }
```

Comment:
why the condition? `homepage` will be removed anyways when equal to `sourceUrl`
```suggestion
        delete dep.homepage;
```

we should always remove it

Reply:
Hey @viceice if you consider the exact case of the bug that you've opened. we shouldn't always remove the `homepage` url. 
For example: 
`homepage: 'https://github.com/org/repo/path/to/something',
sourceUrl: undefined` 
We set `sourceurl` to `homepage` and after massaging `sourceurl` will be `https://github.com/org/repo` so therefore we need to keep `homepage` url set to `https://github.com/org/repo/path/to/something`.
Which is why I've created the `urlPathDepth` function
Happy to provide additional information if needed cause the fix isn't that trivial and will add more tests.

Reply:
then we should remove this here and compare after massaging. if they are equal, remove Homepage